### PR TITLE
feat(diagnostics): support compressed .mcstats streams, add tests for processing stream.

### PR DESCRIPTION
**Pull request by chmeyer-ms**
- tests for ReplayStatsProvider
- optional support for base64 encoded gzip diag file.  MC will always produce this type.

### CODE REVIEW

File: src/stats/replay-stats-provider.ts [hunk 1]
```diff
@@ -138,37 +163,90 @@ export class ReplayStatsProvider extends StatsProvider {
...
-            this._simTickCurrent = statsMessageJson.tick;
+    private _onReadNextStatMessage(rawLine: string) {
+        // first line is the header
+        if (this._replayHeader === undefined) {
+            try {
+                const headerJson = JSON.parse(rawLine);
+                this._replayHeader = headerJson as ReplayStatMessageHeader;
+                this._isGzipped = this._replayHeader.compression === 'gzip';
+            } catch (error) {
+                this._closeReplayStreamReader();
```
Comment (frgarc) [inline src/stats/replay-stats-provider.ts:180]: Any log?

File: src/stats/replay-stats-provider.ts [hunk 2]
```diff
@@ -138,37 +163,90 @@ export class ReplayStatsProvider extends StatsProvider {
...
+        let processedLine = rawLine;
+            try {
+                const buffer = Buffer.from(rawLine, 'base64');
+                if (this._isGzipped) {
+                    processedLine = zlib.gunzipSync(buffer).toString('utf-8');
+                } else {
+                    processedLine = buffer.toString('utf-8');
+                }
+            } catch (error) {
+                this._closeReplayStreamReader();
```
Comment (frgarc) [inline src/stats/replay-stats-provider.ts:195]: Log?

**Review by frgarc — APPROVED** [review]
:shipit:

File: src/stats/replay-stats-provider.ts (hunk 2 above)
Comment (chmeyer-ms) [reply src/stats/replay-stats-provider.ts]: Bad merge (of my own doing), latest has logging.

**Review by frgarc — APPROVED** [review]

**Pull request merged**